src/mut/morphology.cpp

```diff
@@ -388,6 +388,30 @@ void Morphology::applyModifiers(unsigned int modifierFlags)
         modifiers::nrn_order(*this);
 }
 
+std::map<int, std::vector<unsigned int>> Morphology::connectivity()
+{
+    std::map<int, std::vector<unsigned int>> connectivity;
+    for(auto kv: _children)
+    {
+        int node_id = static_cast<int>(kv.first);
+        connectivity[node_id] = std::vector<unsigned int>();
```

Comment:
`map::operator[]` adds the element if not found, so instruction L397 is not required. Besides:
1. the loop over `kv.second` will perform the same access to `connectivity[node_id]` over and over.
1. the assigned vector memory space should be reserved before _push_backing_.
1. the `section` loop variable should be a const reference.
1. use the STL :)

Something like:
```cpp
auto& vec = connectivity[node_id];
vec.reserve(kv.second.size());
std::transform(kv.second.begin(), kv.second.end(), std::back_inserter(vec), [](const auto& section) { return section->id; });
```
Note: the `auto` in the lambda is C++14 only.

Reply:
Thanks. About 1), it's probably optimized anyway by the compiler no ?

Reply:
Yes, in theory, the compilation caches the access because it "sees" that nothing modifies `connectivity` inside it. But I prefer not to bet on that.